packages/core/src/app/shipping/stripeUPE/StripeShippingAddress.tsx

```diff
@@ -83,17 +85,16 @@ const StripeShippingAddress: FunctionComponent<StripeShippingAddressProps> = (pr
     }, [consignments]);
 
     useEffect(() => {
-        if (!isFirstShippingRender && stripeShippingAddress.firstName && !isNewAddress && hasSelectedShippingOptions(consignments)) {
-            setTimeout(() => {
+        if (stripeShippingAddress.firstName && hasSelectedShippingOptions(consignments)) {
+            if (!isFirstShippingRender && !isNewAddress && !isShippingMethodLoading) {
                 if (isStripeLoading && isStripeAutoStep) {
```

Comment:
```suggestion
        const isUsingLinkAndHasShipping = stripeShippingAddress.firstName && hasSelectedShippingOptions(consignments);
        const afterReload = !isFirstShippingRender && !isNewAddress && !isShippingMethodLoading;
        const isLoadingBeforeAutostep =  isStripeLoading && isStripeAutoStep;
        if (isUsingLinkAndHasShipping && afterReload && isLoadingBeforeAutostep) {
```
If possible, please avoid having three conditions nested just to execute a single flow, you can use constants instead.